src/sink/prometheus.rs

```diff
@@ -342,32 +339,62 @@ impl Handler for SenderHandler {
                 // asking for plaintext.
-                let accept: Vec<&str> = req.headers
-                    .get_raw("accept")
-                    .unwrap_or(&[])
-                    .iter()
-                    .map(|x| str::from_utf8(x))
-                    .filter(|x| x.is_ok())
-                    .map(|x| x.unwrap())
-                    .collect();
-                let mut accept_proto = false;
-                for hdr in &accept {
-                    if hdr.contains("application/vnd.google.protobuf;") {
-                        accept_proto = true;
-                        break;
-                    }
-                }
+                // let default_header =
+                //    http::Header::from_bytes(&"Accept"[..], &"text/plain"[..])
+                //        .unwrap();
+                // let accepted: Vec<&str> = request
+                //    .headers()
+                //    .iter()
+                //    .find(|x| {
+                //        x.field == http::HeaderField::from_bytes("Accept").unwrap()
+                //    })
+                //    .unwrap_or(&default_header)
+                //    .value
+                //    .as_str()
+                //    .split(";")
+                //    .collect();
+
+
+                // let mut accept_proto = false;
+                // for accepted_type in &accepted {
+                //    if accepted_type.contains("application/vnd.google.protobuf") {
+                //        accept_proto = true;
+                //        break;
+                //    }
+                // }
                 let reportable = aggr.reportable();
-                let res = if accept_proto {
-                    PROMETHEUS_WRITE_BINARY.fetch_add(1, Ordering::Relaxed);
-                    write_binary(reportable, res)
-                } else {
-                    PROMETHEUS_WRITE_TEXT.fetch_add(1, Ordering::Relaxed);
-                    write_text(reportable, res)
+
+                let mut buffer = Vec::new();
+
+                let buffer = write_text(reportable, buffer).unwrap();
+                let content_encoding = "gzip";
+                let content_type = "text/plain; version=0.0.4";
+                let headers = [
+                    http::Header::from_bytes(&b"Content-Type"[..], content_type)
+                        .unwrap(),
+                    http::Header::from_bytes(
+                        &b"Content-Encoding"[..],
+                        content_encoding,
+                    ).unwrap(),
+                ];
+
+                let response = http::Response::new(
+                    http::StatusCode::from(200),
+                    headers.to_vec(),
+                    &buffer[..],
+                    Some(buffer.len()),
+                    None,
+                );
+
+                match request.respond(response) {
+                    Ok(_) => {
+                        PROMETHEUS_WRITE_TEXT.fetch_add(1, Ordering::Relaxed);
```

Comment:
Previous behaviour was to increment PROMETHEUS_WRITE_TEXT when we attempted to write, uh, text. We can probably remove both WRITE_BINARY and WRITE_TEXT telemetry now that we're all-in on text.

Reply:
Removed both and added PROMETHEUS_REPORT_SUCCESS to compliment PROMETHEUS_REPORT_SUCCESS.